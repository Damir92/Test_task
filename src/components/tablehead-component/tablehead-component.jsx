import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {TableHead, TableRow, TableSortLabel, TableCell} from '@material-ui/core';

import {Columns} from '../../const';
import {getSortingType, getSortingDirection} from '../../store/selectors';
import {setSortDirection, setSortType} from '../../store/action';

const TableHeadComponent = (props) => {
  const {
    sortType,
    sortDirection,
    setSortingType,
    setSortingDirection,
  } = props;

  const handleSortClick = (type) => {
    if (type === sortType) {
      setSortingDirection(!sortDirection);
    } else {
      setSortingType(type);
    }
  };

  return (
    <TableHead>
      <TableRow>
        {Columns.map((item) => (
          <TableCell
            key={item.key}
            padding='none'
            align='center'
          >
            <TableSortLabel
              active={sortType === item.key}
              direction={sortDirection ? `asc` : `desc`}
              onClick={() => {
                handleSortClick(item.key);
              }}
            >
              {item.title}
            </TableSortLabel>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};

TableHeadComponent.propTypes = {
  sortType: PropTypes.string.isRequired,
  sortDirection: PropTypes.bool.isRequired,
  setSortingType: PropTypes.func.isRequired,
  setSortingDirection: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  sortType: getSortingType(state),
  sortDirection: getSortingDirection(state),
});

const mapDispatchToProps = (dispatch) => ({
  setSortingType: (type) => dispatch(setSortType(type)),
  setSortingDirection: (direction) => dispatch(setSortDirection(direction)),
});

export {TableHeadComponent};
export default connect(mapStateToProps, mapDispatchToProps)(TableHeadComponent);
