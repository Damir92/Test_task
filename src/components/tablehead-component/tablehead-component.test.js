import React from 'react';
import {shallow} from 'enzyme';

import {TableHeadComponent} from './tablehead-component';

describe(`TableHead component`, () => {
  it(`TableHeadComponent snapshot`, () => {
    const tree = shallow(
        <TableHeadComponent
          sortType={``}
          sortDirection={true}
          setSortingType={() => {}}
          setSortingDirection={() => {}}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
