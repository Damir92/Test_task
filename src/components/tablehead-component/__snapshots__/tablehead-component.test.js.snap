// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead component TableHeadComponent snapshot 1`] = `
<WithStyles(ForwardRef(TableHead))>
  <WithStyles(ForwardRef(TableRow))>
    <WithStyles(ForwardRef(TableCell))
      align="center"
      key="firstName"
      padding="none"
    >
      <WithStyles(ForwardRef(TableSortLabel))
        active={false}
        direction="asc"
        onClick={[Function]}
      >
        First name
      </WithStyles(ForwardRef(TableSortLabel))>
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="center"
      key="lastName"
      padding="none"
    >
      <WithStyles(ForwardRef(TableSortLabel))
        active={false}
        direction="asc"
        onClick={[Function]}
      >
        Last name
      </WithStyles(ForwardRef(TableSortLabel))>
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="center"
      key="group"
      padding="none"
    >
      <WithStyles(ForwardRef(TableSortLabel))
        active={false}
        direction="asc"
        onClick={[Function]}
      >
        Group
      </WithStyles(ForwardRef(TableSortLabel))>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
</WithStyles(ForwardRef(TableHead))>
`;
